### PR TITLE
Add WasmEdge adapter

### PR DESCRIPTION
**Pull request by harry900831**
WasmEdge is going to add this test suite in the CI and we will maintain our adapter in our repository.

We're comfortable with either decision: adding the adapter here or use the adapter in our repository to run daily runtime validation.

This may be merged after the fixes are resolved on WasmEdge https://github.com/WasmEdge/WasmEdge/issues/2249.

### CODE REVIEW

**Comment by codefromthecrypt** [comment]
@harry900831 there seems a debate about whether or not this should be allowed at all going on in this thread. you can comment if you like https://github.com/WebAssembly/wasi-testsuite/pull/55#issuecomment-1454253271

**Comment by loganek** [comment]
@harry900831 thanks a lot for your change. Would it be possible to host your adapter in the WasmEdge repository, just like we do for Wazero: https://github.com/WebAssembly/wasi-testsuite/pull/55?

Thanks

**Comment by harry900831** [comment]
Done. The adaptor should be in WasmEdge's repo after this be merged: https://github.com/WasmEdge/WasmEdge/pull/2328.
Thanks!

**Review by yamt — APPROVED** [review]
lgtm

File: .github/workflows/daily-runtime-validation.yml [hunk 1]
```diff
@@ -44,6 +44,18 @@ jobs:
...
+        if: matrix.runtime == 'wasmedge'
+        run: |
+          sudo apt install -y software-properties-common cmake libboost-all-dev
+          sudo apt install -y clang
+          git clone https://github.com/WasmEdge/WasmEdge.git
+          cd WasmEdge
+          mkdir -p build && cd build
+          cmake -DWASMEDGE_BUILD_AOT_RUNTIME=OFF .. && make -j
+          echo "$GITHUB_WORKSPACE/WasmEdge/build/tools/wasmedge" >> $GITHUB_PATH
+          cp .github/scripts/wasi-testsuite-adapter.py ../adapters/wasmedge.py
```
Comment (yamt) [inline .github/workflows/daily-runtime-validation.yml:57]: does this mean to test with the latest (unreleased) version of wasmedge?

**Comment by yamt** [comment]
i guess you need a rebase after wazero commit.

**Comment by loganek** [comment]
Do apologize for the delay. The change looks good to me, thanks.

File: .github/workflows/daily-runtime-validation.yml (hunk 1 above)
Comment (harry900831) [reply .github/workflows/daily-runtime-validation.yml:57]: Yes

**Pull request merged**